analysis/study_definition.py

```diff
@@ -1,13 +1,884 @@
+  ),
+  ## Second positive SARS-CoV-2 test
+  covid_test_positive_date2 = patients.with_test_result_in_sgss(
+    pathogen = "SARS-CoV-2", test_result = "positive", returning = "date", date_format = "YYYY-MM-DD", on_or_after = "covid_test_positive_date + 30 days", find_first_match_in_period = True,
+    restrict_to_earliest_specimen_date = True, return_expectations = {"date": {"earliest": "2021-12-20", "latest": "today"}, "incidence": 0.1 },
+  ),
+  ## Positive covid test 30 days prior to positive test [will only apply to patients who first tested positive towards the beginning of the study period]
+  covid_positive_previous_30_days = patients.with_test_result_in_sgss(
+    pathogen = "SARS-CoV-2", test_result = "positive", returning = "binary_flag",  between = ["covid_test_positive_date - 31 days", "covid_test_positive_date - 1 day"],
+    find_last_match_in_period = True, restrict_to_earliest_specimen_date = False, return_expectations = {"incidence": 0.05 },
+  ),
+  # positive test history
+  covid_test_positive_pre_date = patients.with_test_result_in_sgss(
+    pathogen = "SARS-CoV-2", test_result = "positive", find_first_match_in_period = True, on_or_before = "covid_test_positive_date - 30 days",
+    restrict_to_earliest_specimen_date = False, returning = "date",  date_format = "YYYY-MM-DD", return_expectations = {"date": {"earliest": "2021-12-20", "latest": "index_date"}, "incidence": 0.1},
+  ), 
+  ## Onset of symptoms of COVID-19
+  symptomatic_covid_test = patients.with_test_result_in_sgss(
+    pathogen = "SARS-CoV-2", test_result = "any", returning = "symptomatic", on_or_after = "index_date - 5 days", find_first_match_in_period = True,
+    restrict_to_earliest_specimen_date = False, return_expectations={"incidence": 0.1, "category": {"ratios": {"": 0.2, "N": 0.2,"Y": 0.6, } }, },
+  ),
+  covid_symptoms_snomed = patients.with_these_clinical_events(
+    covid_symptoms_snomed_codes, returning = "date", date_format = "YYYY-MM-DD", find_first_match_in_period = True, on_or_after = "index_date - 5 days",
+  ),
+  prior_covid = patients.with_test_result_in_sgss(
+    pathogen = "SARS-CoV-2", test_result = "positive", returning = "binary_flag", on_or_before = "covid_test_positive_date - 1 day", find_last_match_in_period = True, 
+    restrict_to_earliest_specimen_date = True,  return_expectations = {"incidence": 0.9 },
+  ),
+  prior_covid_date = patients.with_test_result_in_sgss(
+    pathogen = "SARS-CoV-2", test_result = "positive", returning = "date", date_format = "YYYY-MM-DD", on_or_before = "covid_test_positive_date - 1 day", find_last_match_in_period = True, 
+    restrict_to_earliest_specimen_date = True, return_expectations = {"date": {"earliest": "2021-12-11", "latest": "today"}, "incidence": 0.9},
+  ),
+  prior_covid_test_type = patients.with_test_result_in_sgss(
```

Comment:
`returning = "case_category"` is only available on positive, earliest specimen date results (so will be missing if this is the second/third time an individual has had covid)

Reply:
i have removed these as actually i do not need them for the analysis